driver-core/src/main/java/com/datastax/driver/core/ThreadingOptions.java

```diff
@@ -0,0 +1,143 @@
+                .setNameFormat(clusterName + "-" + executorName + "-%d")
+                // Back with Netty's thread factory in order to create FastThreadLocalThread instances. This allows
+                // an optimization around ThreadLocals (we could use DefaultThreadFactory directly but it creates
+                // slightly different thread names, so keep we keep a ThreadFactoryBuilder wrapper for backward
+                // compatibility).
+                .setThreadFactory(new DefaultThreadFactory("ignored name"))
+                .build();
+    }
+
+    /**
+     * Builds the generic administration executor, used for tasks such as triggering registered
+     * {@link SchemaChangeListener}s, reacting to node state changes, and metadata updates.
+     * <p/>
+     * The default implementation sets the pool size to the number of available cores.
+     *
+     * @param clusterName the name of the cluster, as specified by
+     *                    {@link com.datastax.driver.core.Cluster.Builder#withClusterName(String)}.
+     * @return the executor.
+     */
+    public ThreadPoolExecutor executor(String clusterName) {
+        ThreadPoolExecutor executor = new ThreadPoolExecutor(
+                NON_BLOCKING_EXECUTOR_SIZE, NON_BLOCKING_EXECUTOR_SIZE,
+                DEFAULT_THREAD_KEEP_ALIVE_SECONDS, TimeUnit.SECONDS,
+                new LinkedBlockingQueue<Runnable>(),
+                threadFactory(clusterName, "worker"));
+        executor.allowCoreThreadTimeOut(true);
+        return executor;
+    }
+
+    /**
+     * Builds the executor used to block on new connections before they are added to a pool.
+     * <p/>
+     * The default implementation uses 2 threads.
+     *
+     * @param clusterName the name of the cluster, as specified by
+     *                    {@link com.datastax.driver.core.Cluster.Builder#withClusterName(String)}.
+     * @return the executor.
+     */
+    public ThreadPoolExecutor blockingExecutor(String clusterName) {
+        ThreadPoolExecutor executor = new ThreadPoolExecutor(
+                2, 2,
+                DEFAULT_THREAD_KEEP_ALIVE_SECONDS, TimeUnit.SECONDS,
+                new LinkedBlockingQueue<Runnable>(),
+                threadFactory(clusterName, "worker"));
+        executor.allowCoreThreadTimeOut(true);
+        return executor;
+    }
+
+    /**
+     * Builds the executor when reconnection attempts will be scheduled.
+     * <p/>
+     * The default implementation uses 2 threads.
+     *
+     * @param clusterName the name of the cluster, as specified by
+     *                    {@link com.datastax.driver.core.Cluster.Builder#withClusterName(String)}.
+     * @return the executor.
+     */
+    public ScheduledThreadPoolExecutor reconnectionExecutor(String clusterName) {
+        return new ScheduledThreadPoolExecutor(2, threadFactory(clusterName, "reconnection"));
+    }
+
+    /**
+     * Builds the executor to handle host state notifications from Cassandra.
```

Comment:
It's also used for debouncing of such notification events, and for idle connections cleanup (again, not sure it's worth mentioning)